File: 2026-01/src/day_01/boj_28278/Main.java
package day_01.boj_28278;

import java.io.*;
import java.util.*;

public class Main {

    static final int MAX = 1000000;
    static int[] stack = new int[MAX];
    static int pos = 0;

    static void push(int x) {
        stack[pos++] = x;
    }

    static int pop() {
        return stack[--pos];
    }

    static int size() {
        return pos;
    }

    static boolean empty() {
        return pos == 0;
    }

    static int top() {
        return stack[pos - 1];
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;

        int n = Integer.parseInt(br.readLine());
        while (n-- > 0) {
            st = new StringTokenizer(br.readLine());
            int q = Integer.parseInt(st.nextToken());

            if (q == 1) {
                int x = Integer.parseInt(st.nextToken());
                push(x);
            } else if (q == 2) {
                if (empty()) {
                    sb.append("-1\n");
                } else {
                    sb.append(pop()).append("\n");
                }
            } else if (q == 3) {
                sb.append(size()).append("\n");
            } else if (q == 4) {
                if (empty()) {
                    sb.append("1\n");
                } else {
                    sb.append("0\n");
                }
            } else {
                if (empty()) {
                    sb.append("-1\n");
                } else {
                    sb.append(top()).append("\n");
                }
            }
        }

        System.out.println(sb);
    }
}

File: 2026-01/src/day_01/boj_28278/Main.cpp
#include <bits/stdc++.h>
using namespace std;

const int MAX = 1000000;
int st[MAX];
int pos = 0;

void push(int x) {
    st[pos++] = x;
}

void pop() {
    pos--;
}

int size() {
    return pos;
}

bool empty() {
    return pos == 0;
}

int top() {
    return st[pos - 1];
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;

    while (n--) {
        int q;
        cin >> q;

        if (q == 1) {
            int x;
            cin >> x;
            push(x);
        } else if (q == 2) {
            if (empty()) {
                cout << -1 << '\n';
            } else {
                cout << top() << '\n';
                pop();
            }
        } else if (q == 3) {
            cout << size() << '\n';
        } else if (q == 4) {
            cout << empty() << '\n';
        } else {
            if (empty()) {
                cout << -1 << '\n';
            } else {
                cout << top() << '\n';
            }
        }
    }
}
